### ISSUE TITLE
Issue when there are properties and explicitly implemented properties with the same name

### ISSUE THREAD

**Issue by kakone** [fluentassertions/fluentassertions#2875]
### Description

Despite the fixes made in version 7.0, this doesn't work correctly when I have properties that have the same name as explicitly implemented interface properties.

Here is the [test project](https://github.com/kakone/FluentAssertionsTestProject) that reproduces the issue.

### Reproduction Steps

```csharp
internal class ClassToTest
{
    public required IMyInterface MyObject { get; set; }
}

internal interface IMyInterface
{
    public TypeCode Type { get; }
}

internal class MyClass : IMyInterface
{
    public string Type { get; } = nameof(TypeCode.Object);

    TypeCode IMyInterface.Type => TypeCode.Object;
}

[TestMethod]
public void TestMethod()
{
    var expected = new ClassToTest() { MyObject = new MyClass() };

    var actual = new ClassToTest() { MyObject = new MyClass() };

    actual.Should().BeEquivalentTo(expected);
}
```

### Expected behavior

No error

### Actual behavior

Error : `Expected property actual.MyObject.Type to be equivalent to TypeCode.Object {value: 1}, but found "Object".`.

### Regression?

It worked in version 6.12.0.

### Are you willing to help with a pull-request?

No

**Comment by IT-VBFK** [comment]
Was this really released with 7.0?

**Comment by dennisdoomen** [comment]
What happens is that FA knows that the `MyObject` property is of type `IMyInterface` so it only "sees" the property of that interface. But when trying to match that to a property on the subject, it only knows the run-time type of the subject, which is `MyClass`. And since that one has two properties, and normal properties have precedence over explicitly implemented or default interface properties, it tries to match it like you see. This is not something we can solve. 

As a workaround, you can use `RespectingRuntimeTypes`. Then the process of finding the properties of the expectation will find both properties.
